docs: add expo router docs (#22381)
expo/expo: docs/pages/router/routing/links.mdx

Next, create a `public/.well-known/apple-app-site-association` file and add the following:

```json title=public/.well-known/apple-app-site-association
{
  "applinks": {
    "apps": [],
    "details": [
      {
        // Example: "QQ57RJ5UTD.com.bacon.app"
        "appID": "{APPLE_TEAM_ID}.{BUNDLE_ID}",
        // All paths that should support redirecting
        "paths": ["*"]
      }
    ]
  }
}
```

Deploy your website to a server that supports HTTPS. See the [hosting guide for more](/docs/deploying/hosting.md).

Finally, install the app on your device and open the website in Safari. You should be prompted to open the app. Expo Router will automatically redirect all web pages to the corresponding native pages.

> Apple docs: [Universal Links](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html)

## Android

Add your website URL to the Android [intent filters](https://docs.expo.dev/versions/latest/config/app/#intentfilters) in your Expo config:

> ```json title=app.json

Reviewer: ```suggestion
```json app.json
```

{
  "expo": {
    "android": {
      "intentFilters": [
        {
          "action": "VIEW",
          "data": {
            "scheme": "https",
            "host": "myapp.com"
          },
          "category": ["BROWSABLE", "DEFAULT"]
        }
      ]
    }
  }
}
```

> Alternatively you can setup intent filters [manually in the AndroidManifest.xml file](https://developer.android.com/training/app-links/verify-android-applinks#add-intent-filters).

> Create a JSON file for the website verification (aka [digital asset links](https://developers.google.com/digital-asset-links/v1/getting-started) file) at `public/.well-known/assetlinks.json` and collect the following information:

Reviewer: ```suggestion
Create a JSON file for the website verification (also known as [digital asset links](https://developers.google.com/digital-asset-links/v1/getting-started) file) at **public/.well-known/assetlinks.json** and collect the following information:
```

> - `package_name`: The Android [application ID](https://docs.expo.dev/versions/latest/config/app/#package) of your app (e.g. `com.bacon.app`). This can be found in the `app.json` file under `expo.android.package`.

Reviewer: ```suggestion
- `package_name`: The Android [application ID](/versions/latest/config/app/#package) of your app (For example, `com.myapp.app`). This is found in the **app.json** under `expo.android.package`.
```

- `sha256_cert_fingerprints`: The SHA256 fingerprints of your app’s signing certificate. This can be obtained in one of two ways:
  1. After building an Android app with EAS Build, run `eas credentials -p android` and select the profile you wish to obtain the fingerprint for. The fingerprint will be listed under `SHA256 Fingerprint`.

>   2. by visiting the [Play Console](https://play.google.com/console/) developer account under `Release > Setup > App Integrity`; if you do, then you'll also find the correct Digital Asset Links JSON snippet for your app on the same page. The value will look like `14:6D:E9:83...`

Reviewer: ```suggestion
  2. Alternatively, by visiting the [Play Console](https://play.google.com/console/) developer account under `Release > Setup > App Integrity`. If you do, then you'll also find the correct Digital Asset Links JSON snippet for your app on the same page. The value will look as `14:6D:E9:83...`.
```

> ```json title=public/.well-known/assetlinks.json

Reviewer: ```suggestion
```json public/.well-known/assetlinks.json
```

[
  {
    "relation": ["delegate_permission/common.handle_all_urls"],
    "target": {
      "namespace": "android_app",
      "package_name": "{package_name}",
      "sha256_cert_fingerprints": [
        // Supports multiple fingerprints for different apps and keys
        "{sha256_cert_fingerprints}"
      ]
    }
  }
]
```

Installing the app will trigger the [Android app verification](https://developer.android.com/training/app-links/verify-android-applinks#web-assoc), which can take up to 20 seconds.

All in-app redirection is handled automatically by Expo Router. You can test your universal links by opening the website in Chrome and clicking on a link. You should be prompted to open the app.

> > Google docs: [App Links](https://developer.android.com/training/app-links)

Reviewer: ```suggestion
> See Google documentation for more information on [App Links](https://developer.android.com/training/app-links).
```

## Debugging

> Expo CLI enables you to test your universal links without deploying a website. Utilizing the [`--tunnel` functionality](https://docs.expo.dev/workflow/expo-cli/#tunneling), you can forward your dev server to a publicly available https URL.

Reviewer: ```suggestion
Expo CLI enables you to test your universal links without deploying a website. Utilizing the [`--tunnel` functionality](/workflow/expo-cli/#tunneling), you can forward your dev server to a publicly available HTTPS URL.
```

1. Set the environment variable `EXPO_TUNNEL_SUBDOMAIN=my-custom-domain` where "my-custom-domain" is a unique string that you will use during development. This will ensure that your tunnel URL is consistent across restarts.
2. Setup universal links as described above, but this time using an Ngrok URL: `my-custom-domain.ngrok.io`
3. Start your dev server with the `--tunnel` flag:

```bash
yarn expo start --tunnel
```

4. Build a development client: `yarn expo run:ios` or `yarn expo run:android`. This will install the development client on your device.

> ## Trouble Shooting

Reviewer: ```suggestion
## Troubleshooting
```
